actions/devenv-k8s-setup-ns/action.yml

```diff
@@ -0,0 +1,177 @@
+
+        # configure namespace
+        if [ -z "$NS_NAME_OVERRIDE" ]; then
+          echo "devenv-namespace=${NS_NAME_PREFIX}-$(uuidgen | cut -c1-6)" | tee -a "$GITHUB_OUTPUT"
+        else
+          echo "devenv-namespace=${NS_NAME_OVERRIDE}" | tee -a "$GITHUB_OUTPUT"
+        fi
+
+        # configure rest of variables
+        echo "pr-number=${PR_NUMBER}" | tee -a "$GITHUB_OUTPUT"
+        echo "workflow-job=${WORKFLOW_JOB}" | tee -a "$GITHUB_OUTPUT"
+        echo "repo-name=${REPO_NAME}" | tee -a "$GITHUB_OUTPUT"
+        echo "branch-name=${BRANCH_NAME}" | tee -a "$GITHUB_OUTPUT"
+        echo "commit-sha=${COMMIT_SHA}" | tee -a "$GITHUB_OUTPUT"
+        echo "workflow-run-number=${WORKFLOW_RUN_NUMBER}" | tee -a "$GITHUB_OUTPUT"
+
+    - name: Checkout devenv repo
+      uses: actions/checkout@v4
+      with:
+        repository: "smartcontractkit/devenv"
+        ref: ${{ inputs.devenv-repo-ref }}
+        path: "${{ github.workspace }}/devenv"
+        token: ${{ inputs.github-token }}
+
+    - name: Create and label Devenv namespace
+      shell: bash
+      env:
+        DEVENV_NAMESPACE:
+          "${{ steps.generate-ns-name.outputs.devenv-namespace }}"
+        NS_NAME_OVERRIDE: ${{ inputs.ns-name-override }}
+      run: |
+        # a valid label must be an empty string or consist of alphanumeric characters,
+        # ‘-’, ‘_’ or ‘.’, and must start and end with an alphanumeric character
+        # (e.g. ‘MyValue’,  or ‘my_value’,  or ‘12345’, regex used for validation is
+        # ‘(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?’)
+        sanitized_branch=$(echo "${{ steps.generate-ns-name.outputs.branch-name }}" \
+          | sed 's/[^a-zA-Z0-9._-]/_/g' | cut -c1-63 | sed 's/[._-]$//')
+
+        # Validate that chainlink-team and chainlink-product inputs adhere to the Kubernetes label format.
+        if [[ ! "${{ inputs.chainlink-team }}" =~ ^[a-zA-Z0-9_.-]+$ ]] || [[ "${{ inputs.chainlink-team }}" -gt 63 ]] || [[ "${{ inputs.chainlink-team }}" =~ ^\. ]] || [[ "${{ inputs.chainlink-team }}" =~ \.$ ]]; then
+          echo "Invalid value for chainlink-team label: ${{ inputs.chainlink-team }}. Must be up to 63 characters, contain only letters, numbers, hyphens, underscores, and dots, and cannot start or end with a dot."
+          exit 1
+        fi
+        if [[ ! "${{ inputs.chainlink-product }}" =~ ^[a-zA-Z0-9_.-]+$ ]] || [[ "${{ inputs.chainlink-product }}" -gt 63 ]] || [[ "${{ inputs.chainlink-product }}" =~ ^\. ]] || [[ "${{ inputs.chainlink-product }}" =~ \.$ ]]; then
+          echo "Invalid value for chainlink-product label: ${{ inputs.chainlink-product }}. Must be up to 63 characters, contain only letters, numbers, hyphens, underscores, and dots, and cannot start or end with a dot."
+          exit 1
+        fi
+
+        overwrite=false
+        echo "Creating ${DEVENV_NAMESPACE} Kubernetes namespace.."
+        if [ -z "$NS_NAME_OVERRIDE" ]; then
```

Comment:
Nit but I would be consistent with using `[[` (preferred) vs. `[` for tests/conditionals.